lib/evss/disability_compensation_form/data_translation.rb

```diff
@@ -0,0 +1,236 @@
+
+        @form_content.compact.to_json
+      end
+
+      private
+
+      def form
+        @form_content['form526']
+      end
+
+      def service_info
+        form['serviceInformation']
+      end
+
+      def veteran
+        form['veteran']
+      end
+
+      def convert_service_info
+        convert_service_periods
+        convert_confinements if service_info['confinements'].present?
+        convert_names if service_info['alternateNames'].present?
+        if service_info['reservesNationalGuardService'].present?
+          service_info['reservesNationalGuardService'] = convert_national_guard_service(
+            service_info['reservesNationalGuardService']
+          )
+        end
+      end
+
+      def convert_service_periods
+        service_info['servicePeriods'].map! do |si|
+          {
+            'serviceBranch' => service_branch(si['serviceBranch']),
+            'activeDutyBeginDate' => si['dateRange']['from'],
+            'activeDutyEndDate' => si['dateRange']['to']
+          }
+        end
+      end
+
+      def convert_confinements
+        service_info['confinements'].map! do |ci|
+          {
+            'confinementBeginDate' => ci['confinementDateRange']['from'],
+            'confinementEndDate' => ci['confinementDateRange']['to'],
+            'verifiedIndicator' => ci['verifiedIndicator']
+          }
+        end
+      end
+
+      def convert_names
+        service_info['alternateNames'].map! do |an|
+          {
+            'firstName' => an['first'],
+            'middleName' => an['middle'],
+            'lastName' => an['last']
+          }.compact
+        end
+      end
+
+      def convert_national_guard_service(reserves_service_info)
+        {
+          'title10Activation' => reserves_service_info['title10Activation'],
+          'obligationTermOfServiceFromDate' => reserves_service_info['obligationTermOfServiceDateRange']['from'],
+          'obligationTermOfServiceToDate' => reserves_service_info['obligationTermOfServiceDateRange']['to'],
+          'unitName' => reserves_service_info['unitName'],
+          'unitPhone' => split_phone_number(reserves_service_info['unitPhone']),
+          'inactiveDutyTrainingPay' => reserves_service_info['inactiveDutyTrainingPay']
+        }.compact
+      end
+
+      def convert_veteran
+        veteran['primaryPhone'] = split_phone_number(veteran['phone'])
+        veteran.delete('phone')
+
+        veteran['mailingAddress'] = convert_mailing_address(veteran['mailingAddress'])
+        if veteran['forwardingAddress'].present?
+          veteran['forwardingAddress'] = convert_mailing_address(veteran['forwardingAddress'])
+        end
+
+        homeless_data = veteran['homelessness']
+        if homeless_data['isHomeless'].present?
+          veteran['homelessness'] = convert_homelessness(homeless_data['pointOfContact'])
+        else
+          veteran.delete('homelessness')
+        end
+      end
+
+      def convert_homelessness(point_of_contact)
```

Comment:
if you do split the methods up could all the homelessness logic be in one method?
```ruby
def convert_homelessness
  data = veteran['homelessness']
  return veteran.delete('homelessness') unless data['isHomeless'].present?
  return veteran['homelessness'] = { 'hasPointOfContact' => false } unless data['pointOfContact'].present?
  veteran['homelessness'] = {
    'hasPointOfContact' => true,
    'pointOfContact' => {
      'pointOfContactName' => data.dig('pointOfContact', 'pointOfContactName'),
      'primaryPhone' => split_phone_number(data.dig('pointOfContact', 'primaryPhone'))
    }
  }
end
```